README.md
# Galette Auto plugin

[![GitHub license](https://img.shields.io/github/license/galette/galette.svg)](https://github.com/galette/plugin-auto/blob/master/COPYING)

### English

A [Galette](https://galette.eu) plugin to manage cars and owners for automobile clubs.

* website: https://galette.eu - https://doc.galette.eu/en/master/plugins/auto.html
* bugs and features: http://bugs.galette.eu/projects/galette-plugin-auto
* mailing lists:
  * users: https://listengine.tuxfamily.org/lists.galette.eu/users/
  * developpers: https://listengine.tuxfamily.org/lists.galette.eu/devel/
* documentation: https://doc.galette.eu/en/master/plugins/auto.html

To use Galette Auto plugin, you'll need a reliable Galette version, and of course the plugin itself by either:

* download latest stable version available from [Galette Auto plugin page](https://doc.galette.eu/en/master/plugins/auto.html)
* use [Galette Auto plugin soure code from repository](http://doc.galette.eu/en/develop/development/git.html) (make sure you install third party dependencies), this solution requires some technical skills

### Français

Un plugin [Galette](https://galette.eu) pour gérer voitures et propriétaires de clubs automobile.

* site web : https://galette.eu - https://doc.galette.eu/fr/master/plugins/auto.html
* bogues et fonctionnalités : http://bugs.galette.eu/projects/galette-plugin-auto
* liste de diffusion :
  * utilisateurs : https://listengine.tuxfamily.org/lists.galette.eu/users/
  * développeurs : https://listengine.tuxfamily.org/lists.galette.eu/devel/
* documentation : https://doc.galette.eu/fr/master/plugins/auto.html

Pour utiliser le plugin Auto pour Galette, vous aurez besoin d'une version adéquate de Galette, ainsi que du plugin lui même :

* télécharger la dernière version stable depuis la [page du  plugin Auto pour Galette](https://doc.galette.eu/en/master/plugins/auto.html)
* utiliser [le code source du plugin Auto pour Galette depuis le dépôt](http://doc.galette.eu/en/develop/development/git.html) (assurez-vous d'installer les biliothèques tierces), cette solution requiert quelques compétences techniques


ChangeLog
* 2021-11-29 - 1.8.0
* Galette 0.9.6 compatible
* Few bugfixes

* 2021-10-23 - 1.7.1
* Few bugfixes

* 2021-07-03 - 1.7.0
- Galette 0.9.5 compatible

* 2020-07-03 - 1.6.0
- Galette 0.9.4 compatible
- Add translations in available core languages (see https://hosted.weblate.org/projects/galette)
- Rely on fontawesome for icons

* 2019-12-01 - 1.5.0
- Galette 0.9.3 compatible
- Remove routes translations
- Fix models sublist
- Fix pagination
- Add an history entry on car creation
- Fix owner selection
- Add hybrid fuel type
- Configure required fields

* 2018-12-02 - 1.4.0
- Galette 0.9.2 compatible

* 2018-04-30 - 1.3.1
- Various fixes
- Add on dashboard

* 2017-12-21 - 1.3.0
- Galette 0.9 compatibility
- Various fixes

* 2016-09-18 - 1.2.0
- Fix removal link
- Fix (un)check javascript
- Fix access on some pages

* 2014-09-14 - 1.1.0
- Galette 0.8 compatibility

* 2013-11-30 - 1.0.0
- First stable release


models.php
<?php

/* vim: set expandtab tabstop=4 shiftwidth=4 softtabstop=4: */

/**
 * Models management
 *
 * This page should be loaded directly, or via ajax.
 * Via ajax, we do not have a full html page, but only
 * that will be displayed using javascript on another page
 *
 * PHP version 5
 *
 * Copyright © 2009-2014 The Galette Team
 *
 * This file is part of Galette (http://galette.tuxfamily.org).
 *
 * Galette is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * Galette is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *  GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Galette. If not, see <http://www.gnu.org/licenses/>.
 *
 * @category  Plugins
 * @package   GaletteAuto
 *
 * @author    Johan Cwiklinski <johan@x-tnd.be>
 * @copyright 2009-2014 The Galette Team
 * @license   http://www.gnu.org/licenses/gpl-3.0.html GPL License 3.0 or (at your option) any later version
 * @version   SVN: $Id: modeles.php 556 2009-03-13 06:48:49Z trashy $
 * @link      http://galette.tuxfamily.org
 * @since     Available since 0.7dev - 2009-09-26
 */

use GaletteAuto\Model;
use GaletteAuto\Brand;

define('GALETTE_BASE_PATH', '../../');
require_once GALETTE_BASE_PATH . 'includes/galette.inc.php';
if ( !$login->isLogged() || (!$login->isAdmin() && !$login->isStaff()) ) {
    header('location: ' . GALETTE_BASE_PATH . 'index.php');
    die();
}

$numrows = $preferences->pref_numrows;
if ( isset($_GET['nbshow']) ) {
    if ( is_numeric($_GET['nbshow']) ) {
        $numrows = $_GET['nbshow'];
    }
}

$is_new = ( get_numeric_form_value('new', '') == 1
        || isset($_POST['donew']) || isset($_GET['donew']) )
    ? true
    : false;
$set = get_form_value('set', null);

//Constants from plugin
require_once '_config.inc.php';

$model = new Model();
$title = ( $is_new ) ? _T("New model", "auto") : _T("Change model '%s'", "auto");

//We have a new or a modified object
if ( get_numeric_form_value('modif', 0) == 1
    || get_numeric_form_value('new', 0) == 1
    && !isset($_POST['cancel'])
) {
    // initialize warnings
    $error_detected = array();
    $warning_detected = array();
    $confirm_detected = array();

    if ( !$is_new && get_numeric_form_value(Model::PK, null) != null ) {
        $model->load(get_numeric_form_value(Model::PK, ''));
    } else if ( !$is_new ) {
        $error_detected[]
            = _T("- No id provided for modifying this record! (internal)", "auto");
    }

    $model->model = get_form_value('model', null);
    $brand_id = get_numeric_form_value('brand', null);

    if ( $brand_id == -1 ) {
        $error_detected[] = _T("- You must select a brand!", "auto");
    } else {
        $model->brand = $brand_id;
    }

    if ( $model->model == '' ) {
        $error_detected[] = _T("- You must provide a value!", "auto");
    }

    if ( count($error_detected) == 0 ) {
        $res = $model->store($is_new);
        if ( !$res ) {
            $error_detected[]
                = _T("- An error occured while saving record. Please try again.", "auto");
        } else {
            header('location:models.php');
        }
    }
} else if ( isset($_POST['cancel']) ) {
    unset($_POST['new']);
    $is_new = false;
    unset($_POST['modif']);
}

//Delete cases. $_GET['sup']} to delete from a single line ;
//$_POST["delete"] to delete from checkboxes
if ( isset($_GET['sup']) || isset($_POST['delete']) ) {
    $del = array();
    if ( isset($_GET['sup']) ) {
        $del[] = $_GET['sup'];
    } else {
        foreach ( $_POST['_sel'] as $supval ) {
            if ( is_numeric($supval) ) {
                $del[] = $supval;
            }
        }
    }

    $model->delete($del);
}

if ( isset($error_detected) ) {
    $tpl->assign('error_detected', $error_detected);
}

//Set the path to the current plugin's templates,
//but backup main Galette's template path before
$orig_template_path = $tpl->template_dir;
$tpl->template_dir = 'templates/' . $preferences->pref_theme;
$tpl->assign('mode', (($is_new) ? 'new' : 'modif'));
$tpl->compile_id = AUTO_SMARTY_PREFIX;

if ( isset( $_GET[Model::PK] ) || $is_new ) {
    if ( !$is_new ) {
        $model->load(get_numeric_form_value(Model::PK, ''));
        $title = str_replace('%s', $model->model, $title);
    }
    if ( isset($_GET['brand']) ) {
        $model->brand = $_GET['brand'];
    }
    $tpl->assign('page_title', $title);
    $tpl->assign('model', $model);
    $b = new Brand();
    $tpl->assign('brands', $b->getList());
    $content = $tpl->fetch('model.tpl', AUTO_SMARTY_PREFIX);
} else {
    $title = _T("Models list", "auto");
    $tpl->assign('page_title', $title);
    $tpl->assign('models', $model->getList());
    $content = $tpl->fetch('models_list.tpl', AUTO_SMARTY_PREFIX);
}
$tpl->assign('content', $content);
//Set path to main Galette's template
$tpl->template_dir = $orig_template_path;
$tpl->display('page.tpl', AUTO_SMARTY_PREFIX);
